crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2068,6 +2179,38 @@ impl<'db> Class<'db> {
     }
 }
 
+/// A utility struct for detecting duplicates in class hierarchies while storing the initial
+/// entry on the stack.
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct Watcher<T: Hash + Eq> {
+    initial: T,
```

Comment:
What's the reason for storing `initial` explicitly over just putting it into the set?

Reply:
We don't have to allocate.